engine/table/src/main/java/io/deephaven/engine/table/impl/RedefinableTable.java

```diff
@@ -25,16 +27,27 @@ protected RedefinableTable(@NotNull final TableDefinition definition, @NotNull f
 
     @Override
     public Table view(Collection<? extends Selectable> selectables) {
+        return viewInternal(selectables, false);
+    }
+
+    @Override
+    public Table updateView(Collection<? extends Selectable> selectables) {
+        return viewInternal(selectables, true);
+    }
+
+    private Table viewInternal(Collection<? extends Selectable> selectables, boolean isUpdate) {
         if (selectables == null || selectables.isEmpty()) {
             return this;
         }
-        final SelectColumn[] columns = SelectColumn.from(selectables);
-        Set<ColumnDefinition<?>> resultColumnsInternal = new HashSet<>();
-        Map<String, ColumnDefinition<?>> resultColumnsExternal = new LinkedHashMap<>();
-        Map<String, ColumnDefinition<?>> allColumns = new HashMap<>(definition.getColumnNameMap());
-        Map<String, Set<String>> columnDependency = new HashMap<>();
-        boolean simpleRetain = true;
-        for (SelectColumn selectColumn : columns) {
+
+        final List<SelectColumn> columns = new ArrayList<>();
```

Comment:
If you do this, you can simplify the rest of the code and revert a lot of the diffs:
```
        final SelectColumn[] columns = Stream.concat(
                isUpdate ? definition.getColumnStream().map(cd -> new SourceColumn(cd.getName())) : Stream.empty(),
                selectables.stream().map(SelectColumn::of)).toArray(SelectColumn[]::new);
```